pkg/features/kube_features.go

```diff
@@ -752,6 +752,13 @@ const (
 	// Enables topology aware hints for EndpointSlices
 	TopologyAwareHints featuregate.Feature = "TopologyAwareHints"
 
+	// owner: @kannon92
+	// kep: https://kep.k8s.io/3939
+	// alpha: v1.28
+
+	// Terminating Pods should count as active for Job and Deployments
```

Comment:
```suggestion
	// Allow customizing pod replacement policy for Jobs.
```
I would also consider adding "... to allow recreating a failed pod only once it reaches the terminal phase.", or something along the lines. I think counting terminating pods as active is more of an implementation detail, but it can stay in the next sentence.

Also, the feature gate does not seem used for Deployments (at least based on its name).